commit 508aa756400e1e614b28b18721359eeee5bbad28
Author: adityaozalwar <adityaozalwar03@gmail.com>
Date:   Fri Jan 9 20:47:06 2026 +0530

    feat: Integrated Coupon Service withing Shopping Cart

diff --git a/src/context/CartContext.jsx b/src/context/CartContext.jsx
index 268434b..6e6f6dd 100644
--- a/src/context/CartContext.jsx
+++ b/src/context/CartContext.jsx
@@ -11,6 +11,7 @@ function CartProvider({ children }) {
   const [finalAmount, setFinalAmount] = useState(0);
   const [cartItems, setCartItems] = useState([]);
   const { currentUser } = useAuth();
+  const [isCouponApplied, setIsCouponApplied] = useState(false);
 
   const getCart = async () => {
     // Add null check here
@@ -23,6 +24,7 @@ function CartProvider({ children }) {
       const data = response.data.response;
       console.log("userid : ", currentUser.userId);
       console.log("data ", data);
+      setIsCouponApplied(data.cartHeader.couponCode != null);
       setDiscountAmount(data.discountAmount);
       setTotalCount(data.totalAmount);
       setFinalAmount(data.finalAmount);
@@ -34,7 +36,14 @@ function CartProvider({ children }) {
 
   return (
     <CartContext.Provider
-      value={{ getCart, totalCount, discountAmount, finalAmount, cartItems }}
+      value={{
+        getCart,
+        totalCount,
+        discountAmount,
+        finalAmount,
+        cartItems,
+        isCouponApplied,
+      }}
     >
       {children}
     </CartContext.Provider>
diff --git a/src/pages/customer/ShoppingCart.css b/src/pages/customer/ShoppingCart.css
index eb7d3bd..8f8cbd6 100644
--- a/src/pages/customer/ShoppingCart.css
+++ b/src/pages/customer/ShoppingCart.css
@@ -56,6 +56,87 @@
   color: #3b82f6;
 }
 
+.coupon-input-wrapper {
+  display: flex;
+  gap: 10px;
+  margin-top: 12px;
+}
+
+.coupon-input-wrapper input {
+  padding: 10px 14px;
+  border-radius: 8px;
+  border: 1px solid #93c5fd;
+  outline: none;
+  font-size: 14px;
+  width: 180px;
+}
+
+.coupon-input-wrapper input:focus {
+  border-color: #2563eb;
+}
+
+.coupon-input-wrapper button {
+  padding: 10px 18px;
+  border-radius: 8px;
+  border: none;
+  background-color: #2563eb;
+  color: white;
+  font-weight: 600;
+  cursor: pointer;
+}
+
+.coupon-input-wrapper button:hover {
+  background-color: #1d4ed8;
+}
+
+.coupon-message {
+  margin-top: 10px;
+  font-size: 14px;
+  font-weight: 500;
+}
+
+.coupon-message.success {
+  color: #15803d;
+}
+
+.coupon-message.error {
+  color: #b91c1c;
+}
+.coupon-input-wrapper button:disabled {
+  opacity: 0.7;
+  cursor: not-allowed;
+}
+.coupon-loading {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  padding: 30px 0;
+  gap: 10px;
+}
+
+.spinner {
+  width: 40px;
+  height: 40px;
+  border: 4px solid #e5e7eb;
+  border-top: 4px solid #2563eb;
+  border-radius: 50%;
+  animation: spin 1s linear infinite;
+}
+
+@keyframes spin {
+  0% {
+    transform: rotate(0deg);
+  }
+  100% {
+    transform: rotate(360deg);
+  }
+}
+.coupon-loading p {
+  color: #6b7280;
+  font-size: 14px;
+  font-weight: 500;
+}
 .sub-total {
   border: 2px solid #059669;
   height: 50%;
@@ -106,4 +187,4 @@
 
 .checkout-btn:active {
   transform: translateY(0);
-}
\ No newline at end of file
+}
diff --git a/src/pages/customer/ShoppingCart.jsx b/src/pages/customer/ShoppingCart.jsx
index 123e5ce..e5056c2 100644
--- a/src/pages/customer/ShoppingCart.jsx
+++ b/src/pages/customer/ShoppingCart.jsx
@@ -8,9 +8,19 @@ import { shopping_cart_api } from "../../utility/url";
 
 function ShoppingCart() {
   const [loading, setLoading] = useState(true);
-  const { getCart, totalCount, discountAmount, finalAmount, cartItems } =
-    useCart();
+  const [couponLoading, setCouponLoading] = useState(false);
+  const {
+    getCart,
+    totalCount,
+    discountAmount,
+    finalAmount,
+    cartItems,
+    isCouponApplied,
+  } = useCart();
   const { currentUser } = useAuth();
+  const [couponCode, setCouponCode] = useState("");
+  const [status, setStatus] = useState(null);
+  const [message, setMessage] = useState("");
 
   useEffect(() => {
     if (currentUser) {
@@ -19,18 +29,81 @@ function ShoppingCart() {
     }
   }, [currentUser]);
 
+  const handleApply = async () => {
+    setCouponLoading(true);
+    setStatus("");
+    setMessage("");
+
+    try {
+      const response = await axios.post(
+        shopping_cart_api + "/api/Cart/applycoupon",
+        {
+          userId: currentUser.userId,
+          couponCode: couponCode,
+        }
+      );
+
+      if (response.data.isSuccess) {
+        setStatus("success");
+        setMessage("Coupon applied successfully!!!");
+
+        setTimeout(() => {
+          setStatus(null);
+          setMessage("");
+        }, 1000);
+        await getCart();
+      } else {
+        setStatus("error");
+        setMessage(response.data.message || "Invalid coupon");
+      }
+    } catch {
+      setStatus("error");
+      setMessage("Something went wrong. Try again.");
+    } finally {
+      setCouponLoading(false);
+    }
+  };
+
+  const handleRemoveCoupon = async () => {
+    setCouponLoading(true);
+    setStatus("");
+    setMessage("");
+
+    try {
+      const response = await axios.post(
+        shopping_cart_api + "/api/Cart/applycoupon",
+        {
+          userId: currentUser.userId,
+          couponCode: "",
+        }
+      );
+
+      if (response.data.isSuccess) {
+        setStatus("success");
+        setMessage("Coupon removed successfully");
+        setCouponCode("");
+        await getCart();
+
+        setTimeout(() => {
+          setStatus(null);
+          setMessage("");
+        }, 1000);
+      } else {
+        setStatus("error");
+        setMessage(response.data.message || "Failed to remove coupon");
+      }
+    } catch {
+      setStatus("error");
+      setMessage("Something went wrong. Try again.");
+    } finally {
+      setCouponLoading(false);
+    }
+  };
+
   const handleCheckout = () => {
     console.log("Proceed to checkout");
   };
 
-  if (loading) {
-    return (
-      <div className="loading-message">
-        <p>Loading products...</p>
-      </div>
-    );
-  }
-
   const removeFromCart = async (productId, userId) => {
     try {
       const response = await axios.delete(
@@ -64,7 +137,34 @@ function ShoppingCart() {
       <div className="side-panel">
         <div className="coupon-window">
           <h3>Coupon Service</h3>
-          <h4>Coming soon...</h4>
+
+          <div className="coupon-input-wrapper">
+            {
+              !isCouponApplied &&
+              <input
+              type="text"
+              placeholder="Enter coupon code"
+              value={couponCode}
+              disabled={isCouponApplied || couponLoading}
+              onChange={(e) => setCouponCode(e.target.value.toUpperCase())}
+            />
+            }
+
+            {!isCouponApplied ? (
+              <button
+                onClick={handleApply}
+                disabled={couponLoading || !couponCode.trim()}
+              >
+                {couponLoading ? "Applying..." : "Apply"}
+              </button>
+            ) : (
+              <button onClick={handleRemoveCoupon} disabled={couponLoading}>
+                {couponLoading ? "Removing..." : "Remove"}
+              </button>
+            )}
+          </div>
+
+          {status && <p className={`coupon-message ${status}`}>{message}</p>}
         </div>
 
         <div className="sub-total">
@@ -72,7 +172,7 @@ function ShoppingCart() {
 
           <div className="summary-row">
             <span>Total Price</span>
-            <span>{totalCount}</span>
+            <span>₹{totalCount}</span>
           </div>
 
           <div className="summary-row">
diff --git a/src/pages/orders/Orders.jsx b/src/pages/orders/Orders.jsx
new file mode 100644
index 0000000..a2c3ab3
--- /dev/null
+++ b/src/pages/orders/Orders.jsx
@@ -0,0 +1,5 @@
+function Orders() {
+  return <>Orders</>;
+}
+
+export default Orders;
diff --git a/src/router/AppRouter.jsx b/src/router/AppRouter.jsx
index 769b2cd..c0dd695 100644
--- a/src/router/AppRouter.jsx
+++ b/src/router/AppRouter.jsx
@@ -44,6 +44,7 @@ function AppRouter() {
           <Route path="inventory" element={<InventoryManagment />} />
         </Route>
         <Route path="shopping-cart" element={<ShoppingCart />} />
+        <Route path="/orders" element={<Orders />} />
         <Route path="/products" element={<Products />} />
         <Route path="/*" element={<PageNotFound />} />
       </Routes>
